modules/core/src/main/java/org/locationtech/jts/io/kml/KMLReader.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (c) 2016 Vivid Solutions.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.locationtech.jts.io.kml;
+
+import org.locationtech.jts.geom.*;
+import org.locationtech.jts.io.ParseException;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.StringReader;
+import java.lang.reflect.Array;
+import java.util.*;
+
+/**
+ * Constructs a {@link Geometry} object from the OGC KML representation.
+ * Works only with KML geometry fragments
```

Comment:
Perhaps say "elements", and mention that it also parses attributes within that element.

Reply:
Fixed